model/user/user.go

```diff
@@ -222,6 +222,33 @@ func (u *DBUser) AddFavoritedProject(identifier string) error {
 	return nil
 }
 
+func (u *DBUser) RemoveFavoriteProject(identifier string) error {
+	if !util.StringSliceContains(u.FavoriteProjects, identifier) {
+		return errors.Errorf("project '%s' does not exist in user's favorites", identifier)
+	}
+	before := u.FavoriteProjects
+
+	update := bson.M{
+		"$pull": bson.M{FavoriteProjectsKey: identifier},
+	}
+	if err := UpdateOne(bson.M{IdKey: u.Id}, update); err != nil {
+		return err
+	}
+
+	for i := len(u.FavoriteProjects) - 1; i >= 0; i-- {
+		if u.FavoriteProjects[i] == identifier {
+			u.FavoriteProjects = append(u.FavoriteProjects[:i], u.FavoriteProjects[i+1:]...)
+		}
+	}
+
+	err := event.LogUserEvent(u.Id, event.UserEventTypeFavoriteProjectsUpdate, before, u.FavoriteProjects)
+	if err != nil {
```

Comment:
do this before you do the update and fail to update if the event log doesn't complete.

Reply:
Taking a step back, do we need to do this at all? Does anyone care when someone updated their favorites?

Reply:
They might care if there are errors in trying to do so. But i do agree that no one cares enough to error out the entire action if the log event fails.

Reply:
My point is that we shouldn't log the event at all. I don't think there's a scenario where someone would want to know this info

Reply:
I think the more confusing thing is that if the eventlog doesn't match what actually happens, it could make things more confusing.

Reply:
i removed the log